CRM/Core/BAO/CustomField.php

```diff
@@ -1647,6 +1647,11 @@ public static function formatCustomField(
               array_values($value)
             ) . CRM_Core_DAO::VALUE_SEPARATOR;
         }
+        // Passing a single value via api-explorer you won't get an array
+        // but a string without separators - CRM-17588
+        elseif (strpos($value, CRM_Core_DAO::VALUE_SEPARATOR) !== 0) {
+          $value = CRM_Core_DAO::VALUE_SEPARATOR . $value . CRM_Core_DAO::VALUE_SEPARATOR;
+        }
       }
       else {
         $value = '';
```

Comment:
This whole thing could be fixed by replacing it with

    if ($value) {
      $value = CRM_Utils_Array::implodePadded($value);
    }
    else {
      $value = '';
    }